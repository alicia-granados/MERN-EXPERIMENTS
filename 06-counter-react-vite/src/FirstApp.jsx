const newMessage= 'Fernando';
const newMessage2 ={
    message : 'hola mundo',
    title : 'Fernando'
}
const saludo = () =>{
    return 'Hola como estas'
}
const FirstApp = () => {
    return (
        <>
        <h2> { 1 + 1 } </h2>
        <p>{newMessage} </p>
        <code>{ JSON.stringify(newMessage2)} </code>
        <p>Soy un subtitulo</p>
        <h1>{saludo()}</h1>
        </>
    )
}

export default FirstApp
