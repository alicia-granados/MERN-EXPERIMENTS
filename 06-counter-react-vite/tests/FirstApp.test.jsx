/** @jest-environment jsdom */
import { render } from "@testing-library/react"
import FirstApp from "../src/FirstApp"

describe('Pruebas en <FirstApp/>', () => {
  /*test('debde de hacer match con el snapshot',() =>{
    const tittle='Hola soy goku'
    const {container} = render(<FirstApp title={tittle}/>)
    expect(container).toMatchSnapshot();

  })
  */
  test('debde de mostrar el titulo en un h1',() =>{
    const tittle='Hola soy goku'
    const {container,getByText, getByTestId } = render(<FirstApp title={tittle}/>)
    expect(getByText(tittle)).toBeTruthy();

    //const h1 = container.querySelector('h1');
    //expect(h1.innerHTML).toBe(tittle) toma espacios
    //expect(h1.innerHTML).toContain(tittle);
    expect(getByTestId('test-title').innerHTML).toContain(tittle);
  })

  test('debde de mostrar el subtitulo enviado por props',() =>{
    const tittle='Hola soy goku'
    const subtitle='Hola soy subtitulo'
    const {getAllByText } = render(
      <FirstApp 
        title={tittle}
        subtitle={subtitle}
      />
    );
    expect(getAllByText(subtitle).length).toBe(2);
  
  })
})
