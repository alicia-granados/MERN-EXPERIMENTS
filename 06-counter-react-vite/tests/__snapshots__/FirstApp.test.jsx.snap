// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debde de hacer match con el snapshot 1`] = `
<div>
  <h1 />
  <h3>
    Hola soy goku
  </h3>
  <h2>
     
    2
     
  </h2>
  <p>
    Fernando
     
  </p>
  <code>
    {"message":"hola mundo","title":"Fernando"}
     
  </code>
  <p>
    Soy un subtitulo
  </p>
  <h1>
    Hola como estas
  </h1>
</div>
`;
